mpa/modules/models/classifiers/sam_classifier.py

```diff
@@ -32,6 +32,8 @@ class SAMImageClassifier(ImageClassifier):
     def __init__(self, task_adapt=None, **kwargs):
         if 'multilabel' in kwargs:
             self.multilabel = kwargs.pop('multilabel')
+        if 'hierarchical' in kwargs:
```

Comment:
Is there a reason why you implemented this on the SAM Classifier? It looks better to use a new classifier that inherits the SAM Classifier.

Reply:
I agree with you. Can I put this work in backlog? I think that work should consider not only hierarchical, but also multilabel.